### PR TITLE
tests: add JA4 test cases -- v3

### PR DESCRIPTION
**Pull request by satta**
Previous PR: #1408

This PR introduces test cases for TLS and QUIC JA4 calculation.

Changes:
* Fix typo in explanatory README.
* Rebase to current master and squash commits.

## Ticket

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/6379

### CODE REVIEW

**Review by inashivb — COMMENTED** [review]
Why do these tests pass if ja4 hasn't been merged yet? 🤔

File: tests/ja4-quic/test.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+requires:
+  min-version: 7.0.0
+  files:
+    - rust/src/quic/ja4.rs
```
Comment (inashivb) [inline tests/ja4-quic/test.yaml:4]: ok. They're likely all skipped for now.

**Comment by satta** [comment]
Yep, that file does not exist until that PR.

File: tests/ja4-quic/test.yaml (hunk 1 above)
Comment (satta) [reply tests/ja4-quic/test.yaml:4]: Yep, that file does not exist until that PR.

**Comment by satta** [comment]
New PR: #1426

**Pull request closed without merging**